bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/navigation/NavigationImpl.java

```diff
@@ -98,23 +98,32 @@ void initModel() {
             items = new ArrayList<>();
             PageManager pageManager = currentPage.getPageManager();
             for (com.adobe.cq.wcm.core.components.models.NavigationItem wcmItem : wcmNavigation.getItems()) {
-                Page page = pageManager.getPage(wcmItem.getPath());
-                if (shouldExpandCatalogRoot(page)) {
-                    expandCatalogRoot(page, items);
-                } else {
-                    String title = wcmItem.getTitle();
-                    String url = wcmItem.getURL();
-                    boolean active = wcmItem.isActive();
-                    NavigationItem item = new PageNavigationItem(null, title, url, active, wcmItem);
-                    items.add(item);
-                }
-
+                addItems(pageManager, null, wcmItem, this.items);
             }
         }
         return items;
     }
 
-    private boolean shouldExpandCatalogRoot(Page page) {
+    private void addItems(PageManager pageManager, AbstractNavigationItem parent,
+        com.adobe.cq.wcm.core.components.models.NavigationItem currentWcmItem, List<NavigationItem> itemList) {
+        Page page = pageManager.getPage(currentWcmItem.getPath());
+        if (shouldExpandCatalogRoot(page)) {
+            expandCatalogRoot(page, itemList);
+        } else {
+            NavigationItem item;
+            if (isCatalogRoot(page)) {
+                item = new CatalogPageNavigationItem(null, page, currentWcmItem);
+            } else {
+                String title = currentWcmItem.getTitle();
```

Comment:
Only cosmetics and optimized by the compiler but these three variables are not really needed. Code can be reduced to one line.